src/protocol/clock/extent.rs

```diff
@@ -0,0 +1,113 @@
+use std::num::{IntErrorKind, TryFromIntError};
+use std::time::Duration;
+
+/// Count fixed sized increments. Provides the product of the increment and its count.
+pub trait IncrementCounter: Sized + Default {
+    type Increment;
+
+    type IncrementsCount;
+    type IncrementIncrementsProduct;
+
+    fn new(size: &Self::Increment, quantity: &Self::IncrementsCount) -> Self;
+
+    fn add(&self, add: Self::IncrementsCount) -> Result<Self, IntErrorKind>;
+    fn sub(&self, sub: Self::IncrementsCount) -> Result<Self, IntErrorKind>;
+
+    fn product_current(&self) -> Result<Option<Self::IncrementIncrementsProduct>, TryFromIntError>;
+    fn product_next(&self) -> Result<Option<Self::IncrementIncrementsProduct>, TryFromIntError>;
+}
```

Comment:
Since the trait is fully abstract now, `Extent`, it could be reused for other situations where you have a count of a unit.

I also think that this trait improves the readability, as it explains the general interface.